script/lib/version-utils.js

```diff
@@ -0,0 +1,63 @@
+const path = require('path')
+const fs = require('fs')
+const semver = require('semver')
+const { getLastMajorForMaster } = require('../get-last-major-for-master')
+const { GitProcess } = require('dugite')
+const { promisify } = require('util')
+
+const readFile = promisify(fs.readFile)
+const gitDir = path.resolve(__dirname, '..', '..')
+
+const getCurrentDate = () => {
+  const d = new Date()
+  const dd = `${d.getDate()}`.padStart(2, '0')
+  const mm = `${d.getMonth() + 1}`.padStart(2, '0')
+  const yyyy = d.getFullYear()
+  return `${yyyy}${mm}${dd}`
+}
+
+const isNightly = v => v.includes('nightly')
+const isBeta = v => v.includes('beta')
+const isStable = v => {
+  v = semver.clean(v)
+  const validFormat = semver.valid(semver.coerce(v))
+  return validFormat ? (validFormat && v.split('.').length === 3) : false
+}
+
+async function nextBeta (v) {
```

Comment:
Do you think unit testing this function and `nextNightly` would be useful? I think I do, and if you agree, we could remove the dependency on Git from this function, e.g.

```javascript
// we'd test this function
function nextUntaggedBeta (currentTags, v) {
  // ...
}

// extract the git operations to 
async function nextBeta (v) {
  const tags = GitProcess.exec(...)
  return nextUntaggedBeta(tags, v)
}
```

or something similar.